ecal/core/src/registration/ecal_registration.cpp

```diff
@@ -106,16 +106,23 @@ namespace eCAL
       return g_descgate()->GetClientInfo(id_, service_info_);
     }
 
-    void GetTopicNames(std::set<std::string>& topic_names_)
+    void GetPublishedTopicNames(std::set<std::string>& topic_names_)
     {
       topic_names_.clear();
 
-      // get publisher & subscriber id sets and insert names into the topic_names set
+      // get publisher id sets and insert names into the topic_names set
       const std::set<STopicId> pub_id_set = GetPublisherIDs();
       for (const auto& pub_id : pub_id_set)
       {
         topic_names_.insert(pub_id.topic_name);
       }
+    }
+
+    void GetSubscribedTopicNames(std::set<std::string>& topic_names_)
+    {
+      topic_names_.clear();
```

Comment:
Why take these as ref if you're going to wipe the input anyway? Assuming the implementation will reuse the previously allocated memory?

```suggestion
    std::set<std::string> GetSubscribedTopicNames()
    {
      std::set<std::string> topic_names_{}
```

Reply:
I'd be more than fine if they just returned the set. I guess we do have a few other functions which take a reference and then return `bool`.
Personally I don't like the style too much (I'd prefer something like `std::expected`, or `std::variant`, but since the core still uses C++ 14 that's out of the question 😞 

Anyways, I guess the biggest problem here is that we're not consistent. `ECAL_API std::set<SServiceId> GetClientIDs();` here we return a set, but `ECAL_API bool GetClientInfo(const SServiceId& id_, ServiceMethodInformationSetT& service_method_info_);` returns bool and takes a ref, and then lastly the functions to just get a name return  `void`.